tensorflow/lite/java/src/main/java/org/tensorflow/lite/Tensor.java

```diff
@@ -311,6 +311,11 @@ private void throwIfTypeIsIncompatible(Object o) {
       return;
     }
     DataType oType = dataTypeOf(o);
+
+    // INT8 and UINT8 have the same string name, "byte"
+    if (oType.toStringName() == dtype.toStringName()) {
```

Comment:
Nit: avoid using `==` in Java string comparisons. We should also have a unit test for DataType to ensure equality.